feat(examples): threejs example

caillef opened the pull request:
Closes #111 

## Introduced changes

- Added ThreeJS example.

https://github.com/dojoengine/dojo.js/assets/20971041/8377c855-7d9e-4672-8ce9-1f6d427e2f35



## Checklist

-   [X] Linked relevant issue
-   [ ] Updated relevant documentation
-   [ ] Added relevant tests
-   [ ] Add a dedicated CI job for new examples
-   [x] Performed self-review of the code

caillef commented:
* Add Lerp
* Use override components

ponderingdemocritus commented on examples/react/react-threejs/src/App.tsx:
```diff
@@ -0,0 +1,30 @@
+import { KeyboardControlsEntry, KeyboardControls } from "@react-three/drei";
+import { UIContainer } from "./ui/components/UIContainer";
+
+export enum Controls {
```
let's move this to dedicated file somewhere

caillef replied:
I removed this, it is never used

ponderingdemocritus commented on examples/react/react-threejs/src/ui/components/UIContainer.tsx:
```diff
@@ -0,0 +1,41 @@
+import { Entity } from "@dojoengine/recs";
+import { getEntityIdFromKeys } from "@dojoengine/utils";
+
+interface TProps {}
```
is this needed? I think this file can be cleaned

caillef replied:
Removed this too

ponderingdemocritus reviewed (commented):
Great work! Lets clean up some of the files

caillef commented:
I've added Lerp, blue cells around the player, moved the Players out of the TileGrid component.
I also added a defineSystem method, but I had to add "world" in the setup.ts return to do so.
I would like your input on this. I did this as I needed to store the current position and the new one for the lerp.

ponderingdemocritus commented on examples/react/react-threejs/index.html:
```diff
@@ -0,0 +1,19 @@
+            rel="stylesheet"
+        />
+
+        <title>Dojo React Vite</title>
```
```suggestion
        <title>Dojo React Three Fibre Starter</title>
```

ponderingdemocritus commented on examples/react/react-threejs/src/gameComponents/UIContainer.tsx:
```diff
@@ -0,0 +1,39 @@
+
+    if (!account) return <></>;
+
+    const handleClick = () => {
```
lets move this to within the onClick - we don't need this extra function

ponderingdemocritus commented on examples/react/react-threejs/src/gameComponents/UIContainer.tsx:
```diff
@@ -0,0 +1,39 @@
+        },
+    } = useDojo();
+
+    if (!account) return <></>;
```
this might not be needed

ponderingdemocritus commented on examples/react/react-threejs/src/gameComponents/Three.tsx:
```diff
@@ -0,0 +1,32 @@
+import { TileGrid } from "./TileGrid";
+import { Players } from "./Players";
+
+const SQUARE_SIZE = 3;
```
move this to a config file

ponderingdemocritus commented on examples/react/react-threejs/src/gameComponents/Players.tsx:
```diff
@@ -0,0 +1,50 @@
+        },
+    } = useDojo();
+
+    const [players, setPlayers] = useState<any>({});
```
it makes sense we convert this to zustand state for local storage, then these players can be accessed else where in the app

caillef commented:
@ponderingdemocritus thanks for the review, I just made the changes.
If you can see what's wrong with the override, I guess the example is ready.
